"attempted duplicate class definition for javassist.CtClassType" when building on JDK7 and running on JDK11

I get the following error when building on OpenJDK 7 and running on 11 (both Zulu). This affects PIT 1.4.6 and 1.4.8 (it was masked by #597 in 1.4.7).
```
org.pitest.util.PitError: 
An error occurred while instantiating class io.github.pr0methean.betterrandom.prng.AesCounterRandomTest: loader 'app' (instance of jdk.internal.loader.Clas
8:02:18 PM PIT >> INFO : MINION : sLoaders$AppClassLoader) attempted duplicate class definition for javassist.CtClassType.

Please copy and paste the information and the complete stacktrace below when reporting an issue
VM : OpenJDK 64-Bit Server VM
Vendor : Azul Systems, Inc.
Version : 11
8:02:18 PM PIT >> INFO : MINION : .0.2+7-LTS
Uptime : 3370
Input -> 
 1 : -Djava.awt.headless=true
 2 : -Djava.security.egd=file:/dev/./urandom
 3 : -javaagent:/tmp/15586417347210670511686154738815976725199667622354.jar
 4 : -Dclassworlds.conf=/usr/share/apache-maven-3.6.0/bin/m2.conf
 5 :
8:02:18 PM PIT >> INFO : MINION :  -Dmaven.home=/usr/share/apache-maven-3.6.0
 6 : -Dlibrary.jansi.path=/usr/share/apache-maven-3.6.0/lib/jansi-native
 7 : -Dmaven.multiModuleProjectDirectory=/home/vsts/work/1/s/betterrandom
BootClassPathSupported : false

	at org.pitest.util.Unchecked.tra
8:02:18 PM PIT >> INFO : MINION : nslateCheckedException(Unchecked.java:20)
	at org.pitest.coverage.execute.CoverageWorker.run(CoverageWorker.java:54)
	at org.pitest.coverage.execute.CoverageMinion.main(CoverageMinion.java:90)
Caused by: org.testng.TestNGException: 
An error occurred while
8:02:18 PM PIT >> INFO : MINION :  instantiating class io.github.pr0methean.betterrandom.prng.AesCounterRandomTest: loader 'app' (instance of jdk.internal.loader.ClassLoaders$AppClassLoader) attempted duplicate class definition for javassist.CtClassType.
	at org.testng.internal.ClassHelper
8:02:18 PM PIT >> INFO : MINION : .createInstance1(ClassHelper.java:435)
	at org.testng.internal.ClassHelper.createInstance(ClassHelper.java:336)
	at org.testng.internal.ClassImpl.getDefaultInstance(ClassImpl.java:125)
	at org.testng.internal.ClassImpl.getInstances(ClassImpl.java:190)
	at 
8:02:18 PM PIT >> INFO : MINION : org.testng.TestClass.getInstances(TestClass.java:95)
	at org.testng.TestClass.initTestClassesAndInstances(TestClass.java:81)
	at org.testng.TestClass.init(TestClass.java:73)
	at org.testng.TestClass.<init>(TestClass.java:38)
	at org.testng.TestRunner.initM
8:02:18 PM PIT >> INFO : MINION : ethods(TestRunner.java:389)
	at org.testng.TestRunner.init(TestRunner.java:271)
	at org.testng.TestRunner.init(TestRunner.java:241)
	at org.testng.TestRunner.<init>(TestRunner.java:167)
	at org.testng.SuiteRunner$DefaultTestRunnerFactory.newTestRunner(Suit
8:02:18 PM PIT >> INFO : MINION : eRunner.java:663)
	at org.testng.SuiteRunner.init(SuiteRunner.java:260)
	at org.testng.SuiteRunner.<init>(SuiteRunner.java:198)
	at org.testng.TestNG.createSuiteRunner(TestNG.java:1295)
	at org.testng.TestNG.createSuiteRunners(TestNG.java:1273)
	at org.tes
8:02:18 PM PIT >> INFO : MINION : tng.TestNG.runSuitesLocally(TestNG.java:1128)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.pitest.testng.TestNGTestUnit.executeInCurrentLoader(TestNGTestUnit.java:88)
	at org.pitest.testng.TestNGTest
8:02:18 PM PIT >> INFO : MINION : Unit.execute(TestNGTestUnit.java:71)
	at org.pitest.coverage.execute.CoverageDecorator.execute(CoverageDecorator.java:50)
	at org.pitest.testapi.execute.containers.UnContainer.execute(UnContainer.java:31)
	at org.pitest.testapi.execute.Pitest.executeTests(
8:02:18 PM PIT >> INFO : MINION : Pitest.java:57)
	at org.pitest.testapi.execute.Pitest.run(Pitest.java:48)
	at org.pitest.coverage.execute.CoverageWorker.run(CoverageWorker.java:51)
	... 1 more
Caused by: java.lang.LinkageError: loader 'app' (instance of jdk.internal.loader.ClassLoaders$A
8:02:18 PM PIT >> INFO : MINION : ppClassLoader) attempted duplicate class definition for javassist.CtClassType.
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1016)
	at java.base/java.security.SecureClassLoa
8:02:18 PM PIT >> INFO : MINION : der.defineClass(SecureClassLoader.java:174)
	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:802)
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:700)
	at java.
8:02:18 PM PIT >> INFO : MINION : base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:623)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(Cla
8:02:18 PM PIT >> INFO : MINION : ssLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	at org.powermock.core.classloader.javassist.ClassPoolFactory.create(ClassPoolFactory.java:33)
	at org.powermock.core.classloader.javassist.JavassistMockClassLoader.<in
8:02:18 PM PIT >> INFO : MINION : it>(JavassistMockClassLoader.java:55)
	at org.powermock.core.classloader.ByteCodeFramework$1.createClassloader(ByteCodeFramework.java:21)
	at org.powermock.core.classloader.MockClassLoaderBuilder.build(MockClassLoaderBuilder.java:58)
	at org.powermock.core
8:02:18 PM PIT >> INFO : MINION : .classloader.MockClassLoaderFactory.createMockClassLoader(MockClassLoaderFactory.java:141)
	at org.powermock.core.classloader.MockClassLoaderFactory.access$100(MockClassLoaderFactory.java:40)
	at org.powermock.core.classloader.MockClassLoaderFactory$1.run(
8:02:18 PM PIT >> INFO : MINION : MockClassLoaderFactory.java:126)
	at org.powermock.core.classloader.MockClassLoaderFactory$1.run(MockClassLoaderFactory.java:122)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at org.powermock.core.classloader.MockClassLoaderFac
8:02:18 PM PIT >> INFO : MINION : tory.createWithPrivilegeAccessController(MockClassLoaderFactory.java:122)
	at org.powermock.core.classloader.MockClassLoaderFactory.createMockClassLoader(MockClassLoaderFactory.java:112)
	at org.powermock.core.classloader.MockClassLoaderFactory.create(Mock
8:02:18 PM PIT >> INFO : MINION : ClassLoaderFactory.java:93)
	at org.powermock.core.classloader.MockClassLoaderFactory.createForClass(MockClassLoaderFactory.java:68)
	at org.powermock.modules.testng.internal.TestNGMockClassLoaderFactory.createClassLoader(TestNGMockClassLoaderFactory.java:
8:02:18 PM PIT >> INFO : MINION : 45)
	at org.powermock.modules.testng.internal.TestClassInstanceFactory.<init>(TestClassInstanceFactory.java:37)
	at org.powermock.modules.testng.internal.PowerMockClassloaderObjectFactory.newInstance(PowerMockClassloaderObjectFactory.java:46)
	at org.power
8:02:18 PM PIT >> INFO : MINION : mock.modules.testng.PowerMockObjectFactory.newInstance(PowerMockObjectFactory.java:43)
	at org.testng.internal.ClassHelper.createInstance1(ClassHelper.java:423)
	... 27 more
```

https://github.com/Pr0methean/BetterRandom/commit/c2febffc9a9e26cc3203ef4542a41c15b65e97eb

https://dev.azure.com/polymorpheus/BetterRandom/_build/results?buildId=540

Full error log: [log_17_540.zip](https://github.com/hcoles/pitest/files/3213959/log_17_540.zip)

--- a/pitest/src/main/java/org/pitest/mutationtest/mocksupport/BendJavassistToMyWillTransformer.java
+++ b/pitest/src/main/java/org/pitest/mutationtest/mocksupport/BendJavassistToMyWillTransformer.java
@@ -3,6 +3,8 @@ package org.pitest.mutationtest.mocksupport;
 import java.lang.instrument.ClassFileTransformer;
 import java.lang.instrument.IllegalClassFormatException;
 import java.security.ProtectionDomain;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.function.Function;
 import java.util.function.Predicate;
 
@@ -10,11 +12,14 @@ import org.objectweb.asm.ClassReader;
 import org.objectweb.asm.ClassVisitor;
 import org.objectweb.asm.ClassWriter;
 import org.pitest.bytecode.FrameOptions;
+import org.pitest.classinfo.ComputeClassWriter;
+import org.pitest.classpath.ClassloaderByteArraySource;
 
 public class BendJavassistToMyWillTransformer implements ClassFileTransformer {
 
   private final Predicate<String> filter;
   private final Function<ClassWriter,ClassVisitor> transformation;
+  private final Map<String, String> computeCache = new ConcurrentHashMap<>();
 
   public BendJavassistToMyWillTransformer(final Predicate<String> filter, Function<ClassWriter,ClassVisitor> transformation) {
     this.filter = filter;
@@ -30,8 +35,10 @@ public class BendJavassistToMyWillTransformer implements ClassFileTransformer {
     if (shouldInclude(className)) {
 
       final ClassReader reader = new ClassReader(classfileBuffer);
-      final ClassWriter writer = new ClassWriter(
-          FrameOptions.pickFlags(classfileBuffer));
+      final ClassWriter writer = new ComputeClassWriter(
+              new ClassloaderByteArraySource(loader), this.computeCache,
+              FrameOptions.pickFlags(classfileBuffer));
+
 
       reader.accept(this.transformation.apply(writer),
           ClassReader.EXPAND_FRAMES);

